### ISSUE TITLE
HostPreflight analyzer with multiple conditions evaluates them all

### ISSUE THREAD

**Issue by xavpaice** [replicatedhq/troubleshoot#1224]
**Bug Description**

When running preflights of type `HostPreflight`, an analyzer with multiple `outcomes` returns a result for all that are true.  This would be OK except that if there's a default fall-through outcome, that returns as well.

**Expected Behavior**

We would expect this to behave like a generic switch statement from top to bottom, returning after the first match.

**Steps To Reproduce**

Create a host preflight:

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: HostPreflight
metadata:
  name: weave
spec:
  collectors:
    - tcpPortStatus:
        collectorName: "Weave Net Control TCP Port Status"
        port: 6783
    - tcpConnect:
        collectorName: "weave 192.168.1.105"
        address: "192.168.1.105:6783"
        timeout: 5s

  analyzers:
    - tcpPortStatus:
        checkName: "Weave Net Control TCP Port Status"
        collectorName: "Weave Net Control TCP Port Status"
        outcomes:
          - fail:
              when: "connection-refused"
              message: Connection to port 6783 was refused. This is likely to be a routing problem since this preflight configures a test server to listen on this port.
          - warn:
              when: "address-in-use"
              message: Another process was already listening on port 6783.
          - fail:
              when: "connection-timeout"
              message: Timed out connecting to port 6783. Check your firewall.
          - fail:
              when: "error"
              message: Unexpected port status
          - pass:
              when: "connected"
              message: Port 6783 is open
          - warn:
              message: Unexpected port status
    - tcpConnect:
        checkName: "weave 192.168.1.105:6783 TCP connection status"
        collectorName: "weave 192.168.1.105"
        outcomes:
          - warn:
              when: "connection-refused"
              message: Connection to weave 192.168.1.105:6783 was refused
          - warn:
              when: "connection-timeout"
              message: Timed out connecting to weave 192.168.1.105:6783
          - warn:
              when: "error"
              message: Unexpected error connecting to weave 192.168.1.105:6783
          - pass:
              when: "connected"
              message: Successfully connected to weave 192.168.1.105:6783
          - warn:
              message: Unexpected TCP connection status for weave 192.168.1.105:6783
```

When this runs (we cannot via the CLI, this is run via kURL [here](https://github.com/replicatedhq/kURL/blob/befa37035988fc097b55c4821ed68894c86961c7/pkg/preflight/preflight.go#L40), the outcome is to return both the `pass` from "connected", as well as the `warn` since there's no associated `when`.  We don't expect to see that last warn if the pass has already been shown.

**Additional Context**

Include the following information.

Came about in a support issue when someone was adding a new node to a cluster, with the Weave add-on.

**Comment by CpuID** [comment]
Struggling to repro this... tried via `kubectl preflight` and `kurl host preflight`, on the latest kURL (`v2023.07.06-0`).

I wonder if I need to go get the `kurl` binary from `v2023.03.21-0` which is what was used in https://github.com/replicated-collab/h2o-replicated/issues/99 - will do some digging.

**Comment by CpuID** [comment]
Hitting this on both kURL versions of the `kurl` binary:

```
Error: decode installer spec: unexpected gvk "troubleshoot.sh/v1beta2, Kind=HostPreflight"
```

Yet if I try and use `troubleshoot.sh/v1beta1` in any spec - I get `Error: decode installer spec: decode spec: no kind "HostPreflight" is registered for version "troubleshoot.sh/v1beta1" in scheme "pkg/runtime/scheme.go:100"`

**Comment by diamonwiggins** [comment]
As @CpuID was uncovering, this was fixed in https://github.com/replicatedhq/troubleshoot/pull/1115 and released in [v0.62.0](https://github.com/replicatedhq/troubleshoot/releases/tag/v0.62.0) of Troubleshoot. This would have been included in kURL version https://github.com/replicatedhq/kURL/releases/tag/v2023.05.08-0 which the customer will need to upgrade to.

**Comment by xavpaice** [comment]
Since this is fixed, I'll close this issue.
